src/sentry/integrations/jira_server/integration.py

```diff
@@ -1097,6 +1099,127 @@ def create_issue(self, data, **kwargs):
         # Immediately fetch and return the created issue.
         return self.get_issue(issue_key)
 
+    def _get_user_by_external_actor(
+        self,
+        client: JiraServerClient,
+        external_issue_key: str,
+        user: RpcUser,
+        logging_context: dict[str, Any],
+    ) -> list[Any] | None:
+        external_actors = ExternalActor.objects.filter(
+            organization_id=self.organization.id,
+            integration_id=self.model.id,
+            provider=ExternalProviders.JIRA_SERVER.value,
+            user_id=user.id,
+        )
+
+        if len(external_actors) > 1:
+            logger.warning(
+                "jira_server.user_external_actor.multiple_actors",
+                extra={
+                    **logging_context,
+                    "user_id": user.id,
+                },
+            )
+            return None
+
+        external_actor: ExternalActor | None = external_actors.first()
+        if external_actor is None:
+            return None
+
+        possible_users = client.search_users_for_issue(
+            external_issue_key, external_actor.external_name
+        )
+
+        for possible_user in possible_users:
+            if possible_user.get("name").lower() == external_actor.external_name.lower():
+                return possible_user
+
+        return None
+
+    def _get_matching_users_by_email(
```

Comment:
This is a copy-paste of the old email lookup logic, with some minor adjustments to match the API that were incorrect previously.

Notably: there's no `username` field in the payload we get from Jira Server, and logging of number of matches has been removed in favor of lookup via `ExternalActor` as the primary way of ensuring assignment sync.